### PR TITLE
EASY-1636 UnsupportedOperationException: empty.reduceLeft

### PR DESCRIPTION
**Pull request by jo-pol**
fixes EASY-1636 UnsupportedOperationException: empty.reduceLeft

#### When applied it will
* throw an exception when the bag-index doesn't find a referenced bag. 
* 
* 

#### How should this be manually tested?

on a fresh deasy (with no bags in the store)

    easy-archive-bag -u user001 -p user001 deposit-1636/doi_10.5061_dryad.8157n/ f6b26ef9-8270-4092-b852-25d339d2bc6b http://localhost:20110

#### related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

**Comment by lindareijnhoudt** [comment]
can't you make this
https://github.com/DANS-KNAW/easy-archive-bag/pull/49/files#diff-c2a6846013675e8fa1d5f1321ada7f75R52
not an Option? Wouldn't that solve our problem with empty refbags?

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 1]
```diff
@@ -67,7 +68,7 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
       case HttpStatus.SC_UNAUTHORIZED =>
         throw UnautherizedException(ps.bagId)
       case _ =>
-        throw new RuntimeException(s"Bag archiving failed: ${ response.getStatusLine }")
+        throw new RuntimeException(s"Bag archiving failed: ${ps.storageDepositService.toString} returned:[ ${ response.getStatusLine } ]. ZippedBag=$zippedBag")
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: @lindareijnhoudt Are the tmp files cleaned up after a while?

Comment (lindareijnhoudt) [reply src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: yes

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 2]
```diff
@@ -81,19 +82,26 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
...
+    val get = new HttpGet(ps.bagIndexService.resolve(s"bag-sequence?contains=$bagId"))
+    get.addHeader("Accept", "text/plain;charset=utf-8")
+    val sw = new StringWriter()
+    val response = http.execute(get)
+    val statusLine = response.getStatusLine
+    if (statusLine.getStatusCode != HttpStatus.SC_OK)
+      throwBagNotFoud(bagId, s"bag index [${get.getURI}] returned ${statusLine.getStatusCode} ${statusLine.getReasonPhrase}")
+    IOUtils.copy(response.getEntity.getContent, sw, "UTF-8")
+    sw.toString match {
+      case s if s.isBlank => throwBagNotFoud(bagId, s"is not found in bag index [${get.getURI}]")
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: Note that the URI duplicates the bag-id of line 102, example of a message:

    Exception in thread "main" java.lang.IllegalStateException: Bag with bag-id 9a840d17-103a-46e3-99a6-abf7ea47642c, pointed to by Is-Version-Of field in bag-info.txt, is not found in bag index [http://localhost:20120/bag-sequence?contains=9a840d17-103a-46e3-99a6-abf7ea47642c]

Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: throwBagNotFoud --> throwBagNotFound

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 3]
```diff
@@ -81,19 +82,26 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
...
-    Try {
-      val http = createHttpClient(ps.bagIndexService.getHost, ps.bagIndexService.getPort, "", "")
-      val get = new HttpGet(ps.bagIndexService.resolve(s"bag-sequence?contains=$bagId"))
-      get.addHeader("Accept", "text/plain;charset=utf-8")
-      val sw = new StringWriter()
-      val response = http.execute(get)
-      if (response.getStatusLine.getStatusCode != HttpStatus.SC_OK) throw new IllegalStateException(s"Error retrieving bag-sequence for bag: $bagId")
-      IOUtils.copy(response.getEntity.getContent, sw, "UTF-8")
-      Some(sw.toString)
+  private def getBagSequence(bagId: BagId)(implicit ps: Parameters): Try[Option[String]] = Try {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: Why do you need the Option here? I don't see where you return a possible None...

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 4]
```diff
@@ -67,7 +68,7 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
       case HttpStatus.SC_UNAUTHORIZED =>
         throw UnautherizedException(ps.bagId)
       case _ =>
-        throw new RuntimeException(s"Bag archiving failed: ${ response.getStatusLine }")
+        throw new RuntimeException(s"Bag archiving failed: ${ ps.storageDepositService } returned:[ ${ response.getStatusLine } ]. ZippedBag=$zippedBag")
```
Comment (lindareijnhoudt) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: could you put this in a logger.error, and leave the original exception msg?

**Pull request merged**